assets/test/e2e/test.js

```diff
@@ -147,22 +141,107 @@ describe('', function() {
       // TODO: validate presented project info, quota and resource info
     });
 
-    it('should view the create page', function() {
+    var goToCreatePage = function() {
       browser.get('/project/test/create');
       expect(element(by.css('h1')).getText()).toEqual("Create Using Your Code");
-      // TODO: validate presented instant apps (load some during e2e)
-    });
+      expect(element(by.css('h3 a')).getText()).toContain("ruby-helloworld-sample");
+    }
 
-    it('should view the create from source repo page', function() {
-      browser.get('/project/test/catalog/images?builderfor=https:%2F%2Fgithub.com%2Fopenshift%2Fnodejs-ex');
+    var fillCreateFromSourceUrl = function(sourceUrl) {
+      element(by.name('from_source_url')).sendKeys(sourceUrl);
+      var nextButton = element(by.buttonText("Next"));
+      browser.wait(protractor.ExpectedConditions.elementToBeClickable(nextButton), 3000);
+      return nextButton.click();
+    }
+
+    var selectBuilderImage = function(builderImageName) {
+      expect(browser.getCurrentUrl()).toMatch(/\/project\/test\/catalog\/images/);
       expect(element(by.css('h1')).getText()).toEqual("Select a builder image");
-      // TODO: validate presented builder images (load some during e2e)
-    });
+      var builderImageLink = element(by.cssContainingText('a', builderImageName));
+      expect(builderImageLink.isPresent()).toBe(true);
+      return builderImageLink.click();
+    }
+
+    var fillAppName = function(builderImageName, appName) {
+      expect(browser.getCurrentUrl()).toMatch(/\/project\/test\/create\/fromimage/);
+      expect(element(by.css('h1')).getText()).toEqual(builderImageName);
+      element(by.name('appname')).clear().sendKeys(appName);
+      return element(by.buttonText("Create")).click();
+    }
 
-    // TODO: show final page of create from source flow
+    function writeScreenShot(data, filename) {
+      var stream = fs.createWriteStream(filename);
 
-    // TODO: show final page of create from template flow
+      stream.write(new Buffer(data, 'base64'));
+      stream.end();
+    }
 
+    var checkApplicationCreated = function(appName) {
+      //browser.pause();
+      expect(browser.getCurrentUrl()).toMatch(/\/project\/test\/overview/); // TODO 11000ms timeout issue
```

Comment:
@jwforres @spadgett I increased the general timeout to 120s now this line passes. Notice however how I dump a screenshot from the running test a couple lines below, and the result is this:

![exception](https://cloud.githubusercontent.com/assets/158611/9095131/0b6f31e6-3b8c-11e5-9b67-099d2d7196fe.png)

"Server connection interrupted". Maybe the browser we use for testing (phantomjs) does not support websockets? Any other clue?